conans/test/integration/settings/test_settings_definition.py

```diff
@@ -0,0 +1,50 @@
+import textwrap
+
+from conans.test.utils.tools import TestClient
+
+
+def test_settings_definitions():
+    c = TestClient()
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        class Pkg(ConanFile):
+            settings = "os"
+            def generate(self):
+                definition = self.settings.get_definition()
```

Comment:
How could users disambiguate (too much ☕ I have big words) or tell a part a `None` from a "not present" value?

Like if a profile was missing cppstd from 1.x and not in the graph then `self.settings.compiler.cppstd.get_definition()` would be None?

Reply:
Not sure what you mean, that gets the range of values (added it to the test):
```python
cppstds = self.settings.compiler.cppstd.get_definition()
self.output.info("CPPSTDS: " + str(cppstds))
````
profile:
```
        [settings]
        compiler=msvc
        compiler.version=192
        compiler.runtime=dynamic
```
```
conan install . -pr=profile -s:b compiler=gcc
...
conanfile.py: CPPSTDS: [None, '14', '17', '20', '23']
```
Python ``None`` will be in the range of possible values

Reply:
Exactly this was the case I was thinking :)